src/pages/search.tsx

```diff
@@ -33,28 +33,39 @@ const Search: React.FC = () => {
     const [currentPage, setCurrentPage] = useState(1);
     const [totalPages, setTotalPages] = useState(1);
 
-    const searchTeacher = async (value: string) => {
-        if (value.trim() && searchedValue.current !== value) {
-            try {
-                const data = await searchByInput(value);
-                setTeachers(data.data);
-                searchedValue.current = value;
-
-                setCurrentPage(data.paging.pageNumber);
-                setTotalPages(data.paging.pageCount);
-            } catch (e) {
-                console.error(e);
-            }
-        }
-    };
-
+    /**
+     * @description Initial loading of data, reflecting the current search state in URL.
+     * Will determine type of search from query params set searched value to the input field.
+     */
```

Comment:
I propose to have some comments and can serve as documentation for places, where logic not exactly straightforward